### ISSUE TITLE
Add Oauth2 providers for Yandex and Mail.ru

### ISSUE THREAD

**Issue by henrypp** [thundernest/k-9#6325]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same feature request.
- [X] I will describe the problem with as much detail as possible.
- [X] This issue only contains a request for one single feature, **not** multiple (related) features.

### App version

6.306

### Problem you are trying to solve

There is a requirement for Oauth2 authorization for Yandex and Mail.ru accounts, as in Thunderbird, but here this is still not available feature.

## Mail.ru:
Documentation: [https://oauth.mail.ru/docs](https://oauth.mail.ru/docs)
Authorization endpoint: [https://oauth.mail.ru/login](https://oauth.mail.ru/login)
Token endpoint: [https://oauth.mail.ru/token](https://oauth.mail.ru/token)

## Yandex:
Documentation: [https://yandex.com/dev/id/doc/dg/imap/index.html](https://yandex.com/dev/id/doc/dg/imap/index.html)
Authorization endpoint: [https://oauth.yandex.com/authorize](https://oauth.yandex.com/authorize)
Token endpoint: [https://oauth.yandex.com/token](https://oauth.yandex.com/token)

**Comment by henrypp** [comment]
I cannot do it myself because i need to provide client id's. It's only on you.

**Comment by henrypp** [comment]
@cketti what about this you can say?

**Comment by cketti** [comment]
Now is a bad time to add support for Russian email providers. Your best bet might be #6152.

**Comment by henrypp** [comment]
Why it is a bad time?

**Comment by vintotal** [comment]
When using Oauth, the autodetected URL has a mistake in it:
https://oauth.yandex.com/authorize?response_type=code&client_id=(caché)*&redirect_uri=http%3A%2F%2Flocalhost&scope=mail%3Aimap_full&login_hint=(caché)%40yandex.ru

It should be « redirect)url », not « recirect_uri ».

**Comment by Hasshu** [comment]
@cketti If I may ask, do you guys risk facing some legal repercussions, or is it about personal convictions only?

**Comment by gabeweb** [comment]
> Now is a bad time to add support for Russian email providers. Your best bet might be [#6152](https://github.com/thunderbird/thunderbird-android/issues/6152).

It makes no sense that Thunderbird for Android still forces me to use app passwords for Yandex Mail in 2026 for political reasons, while the desktop version supports OAuth2 just fine.

**Comment by henrypp** [comment]
@gabeweb 

stupid cunts, sport, software products (and something else) is out of politics, idiots

why K9 supports <s>Gmail</s> Google - <s>Israel</s> USA bombing Iran, it is not funny for all of you?

**Comment by henrypp** [comment]
and one of introduced (SEO) Gmail is Russian, isnt it funny for all of you? a?

**Comment by henrypp** [comment]
Serega Brin him is russian

**Comment by gabeweb** [comment]
> [@gabeweb](https://github.com/gabeweb)
> 
> stupid cunts, sport, software products (and something else) is out of politics, idiots
> 
> why K9 supports Gmail Google - Israel USA bombing Iran, it is not funny for all of you?

Yeah, honestly, I find it ridiculous to lump everyone from a single country together—as if they all think the same, or are all "the enemy," or spies, or whatever.

As a Venezuelan, seeing the West and Europe fighting like toddlers is just cringe. One side shouts "boycott American stuff" and the other goes "ban Chinese and Russian tech," while bragging about European servers that run on Chinese hardware. They don’t even make their own gear anymore. It's embarrassing to watch.

**Comment by Hasshu** [comment]
Guys, while your frustration is understandable, I doubt it's going to help in getting this issue fixed...

**Comment by henrypp** [comment]
@gabeweb 

mozila is OSS foundation, but even he made this bullshit (honestly K9 made by another person, but it still OSS @cketti)

stupid people is over 95% all over the world its a fact

**Comment by gabeweb** [comment]
> Guys, while your frustration is understandable, I doubt it's going to help in getting this issue fixed...

Oops, sorry. Yeah, currently the only workaround (on Android) is generating dedicated (and specific) app passwords, I mean, it's crazy how Yandex ID lets you generate specific passwords for Mail, Calendar, Disk, etc.

**Comment by Alecaddd** [comment]
We do not tolerate any abuse nor any accusation directed to people or personal beliefs.
This behaviour goes against the [Community Participation Guidelines](https://www.mozilla.org/about/governance/policies/participation/) as it not tolerated.

We have more impending issues and architectural problems to tackle and the amount of users using those services is very minor so we're not prioritizing this addition.

**Comment by AdityaGarg8** [comment]
FairEmail uses Oauth2 for yandex and mail.ru
